client/webserver/site/src/js/markets.js

```diff
@@ -198,6 +198,12 @@ export default class MarketsPage extends BasePage {
       }
     })
 
+    bind(document, 'keyup', e => {
+      if (e.key === 'Escape') {
+        closePopups()
+      }
+    })
```

Comment:
`key*` events are triggered on the element that currently in focus (e.g. `input`, `button`), but, in this case, we'd need to catch them on a container element (e.g. the `<form>` wrapping the dialog). But containers can't be in focus so `keyup` etc will bubble up to `document`.

This means, also, that we can have only 1 listener for all dialogs and some callback mechanism to notify the appropriate dialog.  Either that, or `bind`/`unbind` the event handler as the dialogs are shown/hidden, but TBH I don't think it's worth adding all that complexity to make this work on all dialogs.  I still think it's nice to have this on the most frequently used dialog even if the Esc key doesn't work consistently across the board.

What do you think?

Reply:
It doesn't seem problematic to bind to `document` although it's too broad. Let's run with it for now unless buck has thoughts about it.  However, "`bind`/`unbind` the event handler as the dialogs are shown/hidden" doesn't sound like too much complexity to me.

Reply:
The listeners need to be removed 1) when leaving the page, and ideally 2) when the dialog overlay is not visible. 1 needs to be done for sure, but see if you can manage both. 

Getting unbinding to work can sometimes be a pain, because you have to reference the same function object. See https://github.com/decred/dcrdex/blob/08afde3f7bb5d9df5579621ed0e7ae9850424744/client/webserver/site/src/js/charts.js#L86-L98 for one solution

Reply:
To illustrate this need to remove the event listeners, if you put a log in the handler and switch between wallets and markets pages, you can verify that all the old handlers remain bound, causing single key presses to invoke all of the previously bound event listeners

![image](https://user-images.githubusercontent.com/9373513/102903695-4f0b5e80-4436-11eb-8717-783a362b4372.png)

```diff
--- a/client/webserver/site/src/js/markets.js
+++ b/client/webserver/site/src/js/markets.js
@@ -211,6 +211,7 @@ export default class MarketsPage extends BasePage {
     })
 
     bind(document, 'keyup', e => {
+      console.log(e)
```